kernel/sched.c

```diff
@@ -665,6 +666,69 @@ void z_sched_start(struct k_thread *thread)
 	z_reschedule(&sched_spinlock, key);
 }
 
+/**
+ * @brief Halt a thread
+ *
+ * If the target thread is running on another CPU, flag it as needing to
+ * abort and send an IPI (if supported) to force a schedule point and wait
+ * until the target thread is switched out (ISRs will spin to wait and threads
+ * will block to wait). If the target thread is not running on another CPU,
+ * then it is safe to act immediately.
+ *
+ * @param thread Thread to suspend or abort
+ * @param key Current key for sched_spinlock
```

Comment:
Can you more clearly document that this function will release the lock before returning?  That was the source of the last round of review mistakery.

Reply:
Will do.